backend/mora/service/addresses/base.py

```diff
@@ -0,0 +1,92 @@
+#
+# Copyright (c) 2017-2018, Magenta ApS
+#
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+#
+import abc
+import inspect
+
+from ... import exceptions
+from ... import mapping
+from ... import util
+
+ADDRESS_HANDLERS = {}
+
+
+class _AddressHandlerMeta(abc.ABCMeta):
+    """Metaclass for automatically registering handlers."""
+
+    @staticmethod
+    def __new__(mcls, name, bases, namespace):
+        cls = super().__new__(mcls, name, bases, namespace)
+
+        if not inspect.isabstract(cls):
+            cls._register()
+
+        return cls
+
+
+class AddressHandler(metaclass=_AddressHandlerMeta):
+    __slots__ = 'scope', 'prefix', 'value'
+
+    @classmethod
+    def _register(cls):
+        ADDRESS_HANDLERS[cls.scope] = cls
+
+    def __init__(self, value):
+        self.value = value
+
+    @classmethod
+    def from_effect(cls, effect):
+        """Initialize handler from LoRa object"""
+        # Cut off the prefix
+        urn = mapping.SINGLE_ADDRESS_FIELD(effect)[0].get('urn')
+        value = urn[len(cls.prefix):]
+        return cls(value)
+
+    @classmethod
+    def from_request(cls, request):
+        """Initialize handler from MO object"""
+        value = util.checked_get(request, mapping.VALUE, "", required=True)
+        return cls(value)
+
+    def get_urn(self):
+        return self.prefix + self.value
+
+    def get_value(self):
```

Comment:
```suggestion
    @property
    @abstractmethod
    def value(self):
```
1) `get_*` accessors are poor form in Python; that's what `@property` is for.
2) Having a `get_value` method that does nothing other than return `self.value` is entirely redundant; why not simply assign to — and access — `value` instead?

Reply:
1. I'll see what I can do about adding `@property`
2. Because there are cases in the specific handler where `get_value()` _doesn't_ just return the value.